### PR TITLE
signal-cli 0.9.0

### PR DESCRIPTION
**Pull request by ghost**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- Updating to signal-cli v0.9.0
- Providing [fixes](https://github.com/Homebrew/homebrew-core/pull/83322#issuecomment-918945146) for M1 systems

### CODE REVIEW

**Comment by neomsg-git** [comment]
Might need help troubleshooting this.

**Comment by SMillerDev** [comment]
Maybe increase the sleep to 20 seconds?

File: Formula/signal-cli.rb [hunk 1]
```diff
@@ -66,14 +66,23 @@ def install
       # rm originally-embedded libzkgroup library
       system "zip", "-d", zkgroup_jar, "libzkgroup.so"
 
+      # https://github.com/Homebrew/homebrew-core/pull/83322#issuecomment-918945146
+      # this fix is needed until signal-cli updates to zkgroup v0.7.3
+      inreplace "rust-toolchain", "1.41.1", "nightly" if Hardware::CPU.arm?
```
Comment (carlocab) [inline Formula/signal-cli.rb:71]: Can we specify a date on `nightly` so that we get the same Rust compiler across different source builds?

Comment (neomsg-git) [reply Formula/signal-cli.rb:71]: Sure!

**Review by carlocab — APPROVED** [review]
Thanks @neomsg-git!

**Comment by carlocab** [comment]
Oh, I just noticed that your PR branch has merge commits, and is also named `master`. Can you open a new PR with all your commits squashed into a single one [*], and where the PR branch is not named `master`? @BrewTestBot is unable to merge this otherwise.

Thanks!

[*] I think something like 
```
cd $(brew --repo homebrew/core)
git checkout -b signal-cli
git fetch origin
git rebase origin/master
git rebase -i HEAD~10
```
should work here.

**Comment by neomsg-git** [comment]
@carlocab May I ask you to create this pull request then? To be honest I have not looked into git and the Homebrew documentation as much as I would've liked to, due to university. I apologize for the inconvenience...

**Comment by carlocab** [comment]
#85594

**Pull request closed without merging**